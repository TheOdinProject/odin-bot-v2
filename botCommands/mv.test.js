const command = require('./mv');

describe('?mv', () => {
  describe('regex', () => {
    it.each([
      ['?mv'],
      [' ?mv'],
      ['?mv @odin-bot'],
      ['@odin-bot ?mv'],
    ])('correct strings trigger the callback', (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['?v'],
      ['mv'],
      ['?m'],
      ['?mvs'],
      ['```function("?mv", () => {}```'],
      [''],
      [' '],
      [' ?'],
      ['@odin-bot ? mv'],
      ['?m&v'],
      ['?m^v'],
      ['?mv!'],
      ['@odin-bot? mv'],
      ['https:??mv.com'],
    ])("'%s' does not trigger the callback", (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    });

    it.each([
      ['Check this out! ?mv'],
      ['Don\'t worry about ?mv'],
      ['Hey @odin-bot, ?mv'],
      ['?@odin-bot ^ ?me ?mv ?tests$*'],
    ])("'%s' - command can be anywhere in the string", (string) => {
      expect(command.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['@user?mv'],
      ['it\'s about?mv'],
      ['?mvisanillusion'],
      ['?mv?'],
      ['?mv*'],
      ['?mv...'],
    ])("'%s' - command should be its own word/group - no leading or trailing characters", (string) => {
      expect(command.regex.test(string)).toBeFalsy();
    });
  });

  describe('callback', () => {
    it('returns correct output', () => {
      expect(command.cb()).toMatchSnapshot();
    });
  });
});
