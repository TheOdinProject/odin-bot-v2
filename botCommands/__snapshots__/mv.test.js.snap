// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`?mv callback returns correct output 1`] = `"Microverse is using The Odin Project's licensed content without permission. Please read this: https://discord.com/channels/505093832157691914/505093832157691916/781170792095809547"`;
